chat-client/src/client/mynahUi.test.ts

```diff
@@ -0,0 +1,153 @@
+import { before, afterEach } from 'mocha'
+import sinon = require('sinon')
+import { assert } from 'sinon'
+import { createMynahUi, InboundChatApi, handleChatPrompt } from './mynahUi'
+import { Messager, OutboundChatApi } from './messager'
+import { TabFactory } from './tabs/tabFactory'
+import { ChatItemType, MynahUI } from '@aws/mynah-ui'
+
+describe('MynahUI', () => {
+    let messager: Messager
+    let mynahUi: MynahUI
+    let api: InboundChatApi
+    let chatApi: OutboundChatApi
+
+    let getSelectedTabIdStub: sinon.SinonStub
+    let createTabStub: sinon.SinonStub
+    let updateStoreSpy: sinon.SinonSpy
+    let addChatItemSpy: sinon.SinonSpy
+    let onChatPromptSpy: sinon.SinonSpy
+    let onQuickActionSpy: sinon.SinonSpy
+
+    before(() => {
+        chatApi = {
+            sendChatPrompt: sinon.stub(),
+            sendQuickActionCommand: sinon.stub(),
+            tabAdded: sinon.stub(),
+            tabChanged: sinon.stub(),
+            tabRemoved: sinon.stub(),
+            telemetry: sinon.stub(),
+            insertToCursorPosition: sinon.stub(),
+            authFollowUpClicked: sinon.stub(),
+            followUpClicked: sinon.stub(),
+            sendFeedback: sinon.stub(),
+            linkClick: sinon.stub(),
+            sourceLinkClick: sinon.stub(),
+            infoLinkClick: sinon.stub(),
+            uiReady: sinon.stub(),
+        }
+
+        messager = new Messager(chatApi)
+        onChatPromptSpy = sinon.spy(messager, 'onChatPrompt')
+        onQuickActionSpy = sinon.spy(messager, 'onQuickActionCommand')
+
+        const tabFactory = new TabFactory({})
+        createTabStub = sinon.stub(tabFactory, 'createTab')
+        createTabStub.returns({})
+        const mynahUiResult = createMynahUi(messager, tabFactory)
+        mynahUi = mynahUiResult[0]
+        api = mynahUiResult[1]
+        getSelectedTabIdStub = sinon.stub(mynahUi, 'getSelectedTabId')
+        updateStoreSpy = sinon.spy(mynahUi, 'updateStore')
+        addChatItemSpy = sinon.spy(mynahUi, 'addChatItem')
+    })
+
+    afterEach(() => {
+        sinon.resetHistory()
+    })
+
+    it('should handle normal chat prompt', () => {
```

Comment:
Have you checked whether the same is possible to test with real MynahUI (as `chat.test` is doing it)? Exporting `handleChatPrompt` just for the test is not perfect.